models/banana_gun/solana/banana_gun_solana_trades.sql

```diff
@@ -0,0 +1,166 @@
+{{ config(
+    alias = 'trades',
+    schema = 'banana_gun_solana',
+    partition_by = ['block_month'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['blockchain', 'tx_id', 'tx_index', 'outer_instruction_index', 'inner_instruction_index']
+   )
+}}
+
+{% set project_start_date = '2024-01-08' %}
+{% set fee_receiver_1 = '8r2hZoDfk5hDWJ1sDujAi2Qr45ZyZw5EQxAXiMZWLKh2' %}
+{% set fee_receiver_2 = 'Cj297UauzMX64FU9dKJZRUBWszJ7tEWpVheasq4CfATV' %}
+{% set fee_receiver_3 = 'HKMh8nV3ysSofRi23LsfVGLGQKB415QAEfZT96kCcVj4' %}
+{% set fee_receiver_4 = '7tQiiBdKoScWQkB1RmVuML7DBGnR31cuKPEtMM7Vy5SA' %}
+{% set fee_receiver_5 = '4BBNEVRgrxVKv9f7pMNE788XM1tt379X9vNjpDH2KCL7' %}
+{% set wsol_token = 'So11111111111111111111111111111111111111112' %}
+
+WITH
+  allFeePayments AS (
+    SELECT
+      tx_id,
+      'SOL' AS feeTokenType,
+      balance_change / 1e9 AS fee_token_amount,
+      '{{wsol_token}}' AS fee_token_mint_address
+    FROM
+      {{ source('solana','account_activity') }}
+    WHERE
+      {% if is_incremental() %}
+      {{ incremental_predicate('block_time') }}
+      {% else %}
+      block_time >= TIMESTAMP '{{project_start_date}}'
+      {% endif %}
+      AND tx_success
+      AND balance_change > 0
+      AND (
+        address = '{{fee_receiver_1}}'
+        OR address = '{{fee_receiver_2}}'
+        OR address = '{{fee_receiver_3}}'
+        OR address = '{{fee_receiver_4}}'
+        OR address = '{{fee_receiver_5}}'
+      )
+  ),
+  botTrades AS (
+    SELECT
+      trades.block_time,
+      CAST(date_trunc('day', trades.block_time) AS date) AS block_date,
+      CAST(date_trunc('month', trades.block_time) AS date) AS block_month,
+      'solana' AS blockchain,
+      amount_usd,
+      IF(
+        token_sold_mint_address = '{{wsol_token}}',
+        'Buy',
+        'Sell'
+      ) AS type,
+      token_bought_amount,
+      token_bought_symbol,
+      token_bought_mint_address AS token_bought_address,
+      token_sold_amount,
+      token_sold_symbol,
+      token_sold_mint_address AS token_sold_address,
+      fee_token_amount * price AS fee_usd,
+      fee_token_amount,
+      IF(feeTokenType = 'SOL', 'SOL', symbol) AS fee_token_symbol,
+      fee_token_mint_address AS fee_token_address,
+      project,
+      version,
+      token_pair,
+      project_program_id AS project_contract_address,
+      trader_id AS user,
+      trades.tx_id,
+      tx_index,
+      outer_instruction_index,
+      inner_instruction_index
+    FROM
+      {{ ref('dex_solana_trades') }} AS trades
```

Comment:
Is `banana_gun` can be considered as a dex aggregator?

Reply:
feedback in this PR should mimic the feedback in linked PR in description
not adding to `dex.trades`, but reading from it. eventually these spells can form a cross-chain sector level spell for bots